unstable/lens/hash-slice.rkt

```diff
@@ -0,0 +1,26 @@
+#lang racket/base
+
+(provide hash-slice-lens)
+
+(require lens/base/main)
+(module+ test
+  (require rackunit))
+
+(define (hash-slice-lens . ks)
+  (define (get hsh)
+    (for/hash ([k (in-list ks)])
+      (values k (hash-ref hsh k))))
+  (define (set hsh hsh2)
+    (define hsh1
+      (for/fold ([hsh hsh]) ([k (in-list ks)])
+        (hash-remove hsh k)))
+    (for/fold ([hsh hsh1]) ([(k v) (in-hash hsh2)])
+      (hash-set hsh k v)))
+  (make-lens get set))
```

Comment:
I tried using
```racket
(define (hash-slice-lens . ks)
  (apply compound-hash-lens
         (append-map
          (λ (k)
            (list k (hash-ref-lens k)))
          ks)))
```
But it doesn't seem to work properly for setting:
```racket
> (lens-view (hash-slice-lens 'a 'c) (hash 'a 1 'b 2 'c 3))
'#hash((a . 1) (c . 3))
> (lens-set (hash-slice-lens 'a 'c) (hash 'a 1 'b 2 'c 3) (hash 'd 4))
. . ../../../../../../../../../../Applications/Racket/2015-06-08/Racket v6.2.0.4/collects/racket/list.rkt:429:2: map: all lists must have same size; arguments were: #<procedure:list> '(#<procedure:lens-struct> #<procedure:lens-struct>) '(4)
```
But maybe that's because it's trying to enforce the lens laws, like you were talking about in the other comment? But why shouldn't this be allowed?